rotkehlchen/chain/ethereum/modules/polygon_pos_bridge/decoder.py

```diff
@@ -0,0 +1,175 @@
+from typing import TYPE_CHECKING, Any, Final
+
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.polygon.constants import CPT_POLYGON, CPT_POLYGON_DETAILS
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import bridge_match_transfer
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+BRIDGE_ADDRESS: Final = string_to_evm_address('0xA0c68C638235ee32657e8f720a23ceC1bFc77C77')
+ETH_BRIDGE_ADDRESS: Final = string_to_evm_address('0x8484Ef722627bf18ca5Ae6BcF031c23E6e922B30')
+ERC20_BRIDGE_ADDRESS: Final = string_to_evm_address('0x40ec5B33f54e0E8A33A975908C5BA1c14e5BbbDf')
+PLASMA_BRIDGE_ADDRESS: Final = string_to_evm_address('0x401F6c983eA34274ec46f84D70b31C151321188b')
+WITHDRAW_MANAGER_PROXY_ADDRESS: Final = string_to_evm_address('0x2A88696e0fFA76bAA1338F2C74497cC013495922')  # noqa: E501
+POLYGON_STATE_SYNCER_ADDRESS: Final = string_to_evm_address('0x28e4F3a7f651294B9564800b2D01f35189A5bFbE')  # noqa: E501
+STATE_SYNCED_TOPIC: Final = b'\x10?\xed\x9d\xb6^\xac\x19\xc4\xd8p\xf4\x9a\xb7R\x0f\xe0;\x99\xf1\x83\x8eY\x96\xca\xf4~\x9eC0\x83\x92'  # noqa: E501
+EXITED_ERC20_TOPIC: Final = b'\xbba\xbd\x1b&\xb3hL|\x02\x8f\xf1\xa8\xf6\xda\xbc\xac/\xac\x8a\xc5{f\xfak\x1e\xfbn\xde\xab\x03\xc4'  # noqa: E501
+EXITED_ETHER_TOPIC: Final = b'\x0f\xc0\xee\xd4\x1fr\xd3\xdaw\xd0\xf5;\x95\x94\xfcps\xac\xd1^\xe9\xd7\xc56\x81\x9ap\xa6|W\xef<'  # noqa: E501
+START_EXIT_TOPIC: Final = b'\xaaS\x03\xfd\xad\x12:\xb5\xec\xae\xfa\xf6\x917\xbf\x862%x9TmC\xa3\xb3\xdd\x14\x8c\xc2\x87\x9do'  # noqa: E501
+PROCESS_EXIT_TOPIC: Final = b'\xfe\xb2\x00\r\xca>a|\xd6\xf3\xa8\xbb\xb60\x14\xbbT\xa1$\xaa\xc6\xcc\xbfs\xeer)\xb4\xcd\x01\xf1 '  # noqa: E501
+
+
+class PolygonPosBridgeDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.eth = A_ETH.resolve_to_crypto_asset()
```

Comment:
where do you use the self.eth? I don't see it being used. I believe you can remove this logic here